src/migrate-config-options.js

```diff
@@ -0,0 +1,87 @@
+'use babel'
+
+// These are cases where the old setting should be directly moved to the new setting.
+const directMoveMigrations = [
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.disableWhenNoEslintConfig',
+    new: 'linter-eslint.disabling.disableWhenNoEslintConfig',
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.ignoreFixableRulesWhileTyping',
+    new: 'linter-eslint.disabling.ignoreFixableRulesWhileTyping'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.rulesToDisableWhileFixing',
+    new: 'linter-eslint.disabling.rulesToDisableWhileFixing'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.rulesToSilenceWhileTyping',
+    new: 'linter-eslint.disabling.rulesToSilenceWhileTyping'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.disableEslintIgnore',
+    new: 'linter-eslint.advanced.disableEslintIgnore'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.disableFSCache',
+    new: 'linter-eslint.advanced.disableFSCache'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.showRuleIdInMessage',
+    new: 'linter-eslint.advanced.showRuleIdInMessage'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.eslintrcPath',
+    new: 'linter-eslint.advanced.eslintrcPath'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.advancedLocalNodeModules',
+    new: 'linter-eslint.advanced.advancedLocalNodeModules'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.useGlobalEslint',
+    new: 'linter-eslint.global.useGlobalEslint'
+  },
+  {
+    /* Added November, 2017 */
+    old: 'linter-eslint.globalNodePath',
+    new: 'linter-eslint.global.globalNodePath'
+  },
+]
+
+function migrateConfigOptions() {
+  /**
+   * FIXME: Deprecated eslintRulesDir{String} option in favor of
+   * eslintRulesDirs{Array<String>}. Remove in the next major release,
+   * in v8.5.0, or after 2018-04.
+   */
+  const oldRulesdir = atom.config.get('linter-eslint.eslintRulesDir')
```

Comment:
Do we want to integrate this into the data structure above as a `migrationFunc` or something like that? If it exists it gets called instead of the default migration function.

Reply:
Yeah I can play around with that.